crates/rspack_binding_options/src/options/raw_module/js_loader.rs

```diff
@@ -416,6 +411,80 @@ pub async fn run_builtin_loader(
   JsLoaderContext::try_from(&mut cx).map_err(|e| Error::from_reason(e.to_string()))
 }
 
+pub struct AdditionalDataRef {
```

Comment:
First of all, this implementation is great. However it's not error-prone. I would recommend you to use `napi::Ref<()>` as the type to store the `AdditionalData`. You may wish to take [this](https://github.com/web-infra-dev/rspack/blob/ee1a6893f9a248a9378c2f1ac08b572798fc5334/crates/rspack_fs_node/src/node.rs#L5) as an example.

Reply:
I use the `napi::Ref<()>` to hold the JS value, but it doesn't implement neither `Copy` nor `Clone` trait, so I wrap it with `Arc<Mutex<_>>`. And I need to implement `Send` and `Sync` for it due to `napi::Env` is not thread safe.

The `AdditionalDataRef` is renamed to `JsValueRef` and moved to `src/options/raw_module/js_value_ref.rs`.